node/src/components/api_server.rs

```diff
@@ -77,7 +81,10 @@ impl<REv> ReactorEventT for REv where
 }
 
 #[derive(Debug)]
-pub(crate) struct ApiServer {}
+pub(crate) struct ApiServer {
+    /// Channel sender to pass event-stream data to the event-stream server.
+    sse_data_sender: UnboundedSender<SseData>,
```

Comment:
I see your point, but I think it's safe in this case.  The events being generated by the node (the inputs to this channel) shouldn't be coming at a high rate for any sustained length of time.  The other end of the channel has very little work to do upon receiving the channel's messages - they're simply added to the ring buffer and fired off down the broadcast channel (which _is_ limited in size).  So I expect this means the unbounded channel here shouldn't ever overfill.

This is all just guesswork of course - I could well be wrong, or things may change in the future!  However... I also think it's quite important that the node reports all events it should.  If we have that premise, then buffering events in some other place until a bounded sender is able to actually send them would only be shifting the problem I believe?

Reply:
Definitely, if you hit the bound of a channel, it will block, and just shift that issue to another area. I questioned it more for my own education than as a genuine concern.